### PR TITLE
feat: add migration for sfc components

### PR DESCRIPTION
**Pull request by eneajaho**
Closes https://github.com/ngxtension/ngxtension-platform/issues/362

### CODE REVIEW

File: docs/src/content/docs/utilities/Migrations/sfc-migration.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+### How it works?
+
+The moment you run the schematics, it will look for all the components in your project and will convert them to SFC components.
+
+- It will move the template from the `templateUrl` to the `template` property.
+- The maximum lines length for the template is set to 200 lines. If the template has more than 200 lines, it will be skipped.
+
+In order to change the maximum line length, you can pass the `--max-inline-template-lines` param to the schematics.
+
+````bash
```
Comment (michael-small) [inline docs/src/content/docs/utilities/Migrations/sfc-migration.md:21]: extra ```bash that throws off formatting

File: libs/plugin/src/generators/convert-to-sfc/generator.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+			const decoratorArg = applicableDecorator.getArguments()[0];
+			if (Node.isObjectLiteralExpression(decoratorArg)) {
+				decoratorArg
+					.getChildrenOfKind(SyntaxKind.PropertyAssignment)
+					.forEach((property) => {
+						const decoratorPropertyName = property.getName();
+						if (decoratorPropertyName === 'templateUrl') {
+							const dir = dirname(sourcePath);
+							const templatePath = joinPathFragments(
```
Comment (ilirbeqirii) [inline libs/plugin/src/generators/convert-to-sfc/generator.ts:109]: Is it possible to create a reusable function for preparing "templatePath", and then reading its content? I also noticed a similar code in the "self-closing tags" schematic.

Comment (eneajaho) [reply libs/plugin/src/generators/convert-to-sfc/generator.ts:109]: Good idea. Will do as a refactoring step.

**Review by nartc — APPROVED** [review]

**Review by pavankjadda — COMMENTED** [review]
LGTM

**Pull request merged**